instruments/toptica/topmode.py

```diff
@@ -426,7 +417,7 @@ def firmware(self):
         :return: The firmware version of the charm controller
         :type: `str`
         """
-        firmware = tuple(map(int, self.reference("fw-ver").split(".")))
+        firmware = [int(i) for i in self.reference("fw-ver").split(".")]
```

Comment:
We should return a tuple (firmware versions doesn't need to be mutable). Does this follow the usual X.Y.Z formatting? If you think they might come out with updates that changes this (say, from X.Y.Z to X.Y) we might want to enforce a minimum length here by appending zeros until the length is 3. I think we did that for some of the Qubitekk instruments iirc.